x/milestone/abci/abci.go

```diff
@@ -155,11 +157,17 @@ func GetMajorityMilestoneProposition(
 
 		prop := voteExtension.MilestoneProposition
 		for i, blockHash := range prop.BlockHashes {
+			blockTd := prop.BlockTds[i]
```

Comment:
We need to add validation inside `ValidateMilestoneProposition` that len(BlockTds) == len(BlockHashes).